eng/common/TestResources/New-TestResources.ps1

```diff
@@ -546,12 +560,28 @@ try {
             }
 
             $outputFile = "$($templateFile.originalFilePath).env"
+            if ($env:ENV_FILE) {
```

Comment:
LOL... OutFile is a switch so then the question is why don't we support this via a parameter instead of an ENV variable here?

Reply:
It is a switch, yes, but the output file shouldn't have to be a path: just always use a predictable path that our test infrastructure code (like we have in .NET) knows how to find regardless of platform. How it decrypts the file would be platform-dependent, though, but that's an exercise for language teams. Just be consistent.

Reply:
There are reasons we set it up this way initially, which I'll explain below, but I think there's a solution. @ckairen I hadn't thought of it before, but we may be able to use `mountPath` and/or `subPath` (see [here](https://kubernetes.io/docs/concepts/storage/volumes/#example-configuration-secret-downwardapi-configmap)) as a way of re-using the same volume for configmap input and deployment outputs.

For a given stress test deployment, we first run an "init container" that performs the azure deployment. This container has three volume mounts:

1. static secrets (service principal credentials), mounted to `/mnt/secrets/static`
2. ARM templates mounted from a "configmap" (kubernetes resource type for storing plaintext objects) to `/mnt/azure`
3. An output directory for sharing secrets from the init container to the main test container that a stress test user owns, mounted to `/mnt/outputs`

With the default script behavior, the service directory is set to `/mnt/azure` so it knows where to look for the ARM template, and would normally output back to the same service directory `/mnt/azure`. The stress test container specs do not reference the ARM template configmap, because they have no need for it. They only reference the shared volume mount at `/mnt/outputs` in order to receive files generated by the init containers, hence why we initially needed a separate output path not governed by the script's restrictions/convention.